trio/io/types.py

```diff
@@ -0,0 +1,72 @@
+from functools import partial
+
+import trio
+
+
+__all__ = ['AsyncRawIOBase', 'AsyncBufferedIOBase', 'AsyncTextIOBase']
+
+
+def _method_factory(cls, meth_name):
+    async def wrapper(self, *args, **kwargs):
+        meth = getattr(self._file, meth_name)
+        func = partial(meth, *args, **kwargs)
+        return await trio.run_in_worker_thread(func)
+
+    wrapper.__name__ = meth_name
+    wrapper.__qualname__ = '.'.join((__name__,
+                                     cls.__name__,
+                                     meth_name))
```

Comment:
I don't think so; `__name__` is `trio.io.types` (the module name), so it becomes something like `trio.io.types.*IOBase.close`.